subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -295,13 +296,22 @@ uint8_t ll_df_set_cl_cte_tx_enable(uint8_t adv_handle, uint8_t cte_enable)
 			return BT_HCI_ERR_CMD_DISALLOWED;
 		}
 
-		err = ull_adv_sync_pdu_set_clear(adv, 0,
+		err = ull_adv_sync_pdu_alloc(adv, 0,
+					     ULL_ADV_PDU_HDR_FIELD_CTE_INFO,
+					     NULL, &pdu_prev, &pdu, &ter_idx);
+		if (err) {
+			return err;
+		}
+
+		err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu, 0,
 						 ULL_ADV_PDU_HDR_FIELD_CTE_INFO,
-						 NULL, &ter_idx);
+						 NULL);
 		if (err) {
 			return err;
 		}
 
+		lll_adv_sync_data_enqueue(lll_sync, ter_idx);
```

Comment:
`lll_adv_sync_data_enqueue` is called once, just after end of if-else statement. It covers both parts of condition. This call is not needed here.